hash/hasher_bcrypt_aes.go

```diff
@@ -0,0 +1,88 @@
+package hash
+
+import (
+	"bytes"
+	"context"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"io"
+
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/bcrypt"
+	"golang.org/x/crypto/sha3"
+
+	"github.com/ory/kratos/driver/config"
+)
+
+var BcryptAESAlgorithmId = []byte("bcryptAes")
+var NoAESKeyError = errors.New("encode error: no AES keys were configured")
+
+type BcryptAES struct {
+	c BcryptAESConfiguration
+}
+
+type BcryptAESConfiguration interface {
+	config.Provider
+}
+
+func NewHasherBcryptAES(c BcryptAESConfiguration) *BcryptAES {
+	return &BcryptAES{c: c}
+}
+
+func (h *BcryptAES) aes256Encrypt(data, key []byte) ([]byte, error) {
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+
+	gcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return nil, err
+	}
+
+	nonce := make([]byte, gcm.NonceSize())
+	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	hash := append(nonce, gcm.Seal(nil, nonce, data, nil)...)
+	encoded := make([]byte, hex.EncodedLen(len(hash)))
+	hex.Encode(encoded, hash)
+
+	return encoded, nil
+}
```

Comment:
https://github.com/gtank/cryptopasta/blob/master/encrypt.go

Reply:
Done in c5b5e31e